docs/features/hooks.md

```diff
@@ -2,6 +2,117 @@
 
     Check out the [IDOM Core docs](https://idom-docs.herokuapp.com/docs/reference/hooks-api.html?highlight=hooks) on hooks!
 
+## Use Query
+
+The `use_query` hook is used make ORM queries.
+
+=== "components.py"
+
+    ```python
+    from example_project.my_app.models import TodoItem
+    from idom import component, html
+    from django_idom.hooks import use_query
+
+    def get_items():
+        return TodoItem.objects.all()
+
+    @component
+    def todo_list():
+        item_query = use_query(get_items)
+
+        if item_query.loading:
+            rendered_items = html.h2("Loading...")
+        elif item_query.error:
+            rendered_items = html.h2("Error when loading!")
+        else:
+            rendered_items = html.ul(html.li(item, key=item) for item in item_query.data)
+
+        return rendered_items
+    ```
+
+=== "models.py"
+
+    ```python
+    from django.db import models
+
+    class TodoItem(models.Model):
+        text = models.CharField(max_length=255)
+    ```
+
+??? question "Can I make ORM calls without hooks?"
+
+    Due to Django's ORM design, database queries must be deferred using hooks. Otherwise, you will see a `SynchronousOnlyOperation` exception.
+
+    This may be resolved in a future version of Django with a natively asynchronous ORM.
+
+??? question "What is an "ORM"?"
+
+    A Python **Object Relational Mapper** is an API for your code to access a database.
+
+    See the [Django ORM documentation](https://docs.djangoproject.com/en/dev/topics/db/queries/) for more information.
+
+## Use Mutation
+
+The `use_mutation` hook is used to modify ORM objects.
+
+=== "components.py"
+
+    ```python
+    from example_project.my_app.models import TodoItem
+    from idom import component, html
+    from django_idom.hooks import use_mutation
+
+    def add_item(text: str):
+        TodoItem(text=text).save()
+
+    @component
+    def todo_list():
+        item_mutation = use_mutation(add_item)
```

Comment:
A better example here would be to display the todo items in this component too. You could then demonstrate `use_mutation(add_item, refetch=get_items)`.

The purpose of refetch is to re-run any queries that might have been impacted by a mutation. For example, in this case, after running `add_item` to the database, your prior `get_items` query would be stale and should be re-executed to get the latest items. This is especially useful if `get_items` was used in a different component since, `use_mutation(add_items, refetch=get_items)` will automatically tell components that used `use_query(get_items)` to re-render regardless of where they are in your layout.

Reply:
I planned on having that more expansive example within a question collapse. I think it's too much info to give at once if that's at a top level.

Reply:
Got it. That makes sense.